### PR TITLE
Fix panic when progid is the last token by removing unwrap

### PR DESCRIPTION
**Pull request by MidasLamb**
Also first suggestion of how to go about testing for panics found by fuzz testing.

### CODE REVIEW

File: src/parse/value/css_function.rs [hunk 1]
```diff
@@ -64,7 +64,12 @@ impl<'a> Parser<'a> {
 
     pub(super) fn parse_progid(&mut self) -> SassResult<String> {
         let mut string = String::new();
-        let mut span = self.toks.peek().unwrap().pos();
+        let mut span = match self.toks.peek() {
+            Some(token) => token.pos(),
+            None => {
+                return Err(("Unexpected end of tokenstream.", self.span_before).into());
```
Comment (connorskees) [inline src/parse/value/css_function.rs:70]: The `dart-sass` error message in this case is

```
Error: expected "(".
```

Could we change the message to that (exluding the `Error:` bit)?

File: tests/fuzz-tests.rs [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#[macro_use]
+mod macros;
+
+no_panic!(progid, "j:progid:");
```
[the remark this answers is not in the document]
Comment (connorskees) [reply tests/fuzz-tests.rs:4]: I am hoping to remove this bug in an upcoming rewrite!

**Comment by connorskees** [comment]
This is a fantastic catch! I've left some comments about test structuring. Thanks for finding this!

File: tests/macros.rs [hunk 3]
```diff
@@ -50,6 +50,20 @@ macro_rules! error {
     };
 }
 
+/// Verify that the input does not cause a panic.
+/// Useful for when testing input found through fuzzing.
+#[macro_export]
+macro_rules! no_panic {
```
Comment (pickfire) [inline tests/macros.rs:56]: Isn't the default test already means no panic?

File: tests/fuzz-tests.rs [hunk 4]
```diff
@@ -0,0 +1,4 @@
+#[macro_use]
```
[the remark this answers is not in the document]
Comment (MidasLamb) [reply tests/fuzz-tests.rs]: Done!

File: tests/fuzz-tests.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (MidasLamb) [reply tests/fuzz-tests.rs]: Changed to the requested case after verifying it triggered the same panic.

[the remark this answers is not in the document]
Comment (MidasLamb) [reply tests/fuzz-tests.rs]: I wasn't sure about the approach, so moved it to `error!` as suggested

File: src/parse/value/css_function.rs (hunk 1 above)
Comment (MidasLamb) [reply src/parse/value/css_function.rs]: Wasn't sure if you wanted to also copy the error message exactly, now it's clear you want to have the exact same behavior as dart-sass, down to the error messages. Will keep this in my going forward!

File: tests/macros.rs (hunk 3 above)
Comment (MidasLamb) [reply tests/macros.rs]: I wanted to make clear that this macro verifies nothing, it doesn't check if the outcome is `Ok` or `Err`, only that it doesn't panic. Which is indeed normal test behavior, so the macro only calls the function with the default options and nothing goes on inside.

**Comment by connorskees** [comment]
Thanks for this!

**Pull request merged**